### PR TITLE
Fix 139

### PR DESCRIPTION
**Pull request by gianarb**
Rebase #139

### CODE REVIEW

File: src/App.php [hunk 1]
```diff
@@ -133,13 +133,15 @@ public function run($request = null, $response = null)
...
-     * @param PennyEventInterface $event
+     * @param RouteInfoInterface $routeInfo
+     * @param EventInterface $event
      *
      * @throws RuntimeException if dispatch does not return RouteInfo object.
      */
-    private function handleRoute($routeInfo, PennyEventInterface $event)
-    {
+    private function handleRoute(
+        RouteInfoInterface $routeInfo,
```
Comment (samsonasik) [inline src/App.php:142]: remove RouteInfoInterface type hint

File: src/App.php [hunk 2]
```diff
@@ -133,13 +133,15 @@ public function run($request = null, $response = null)
     /**
      * Handle Route.
      *
-     * @param mixed $routeInfo
-     * @param PennyEventInterface $event
+     * @param RouteInfoInterface $routeInfo
```
Comment (samsonasik) [inline src/App.php:136]: @param mixed $routeInfo

**Comment by samsonasik** [comment]
I will re-create PR for it.

File: src/App.php (hunk 2 above)
Comment (gianarb) [inline src/App.php:136]: Why? The dispatch method triggers exceptions or return RouteInfo object

File: src/App.php (hunk 1 above)
Comment (gianarb) [inline src/App.php:142]: Why?

**Comment by samsonasik** [comment]
@gianarb you're wrong on rebase

File: src/App.php (hunk 2 above)
Comment (samsonasik) [inline src/App.php:136]: but the exception won't be executed, as the type hint is RouteInfoInterface

Comment (samsonasik) [inline src/App.php:136]: this is the internal method that handle RouteInfoInterface check, so param need accept not a RouteInfoInterface, so the RuntimeException will be executable!

**Pull request merged**